hswindow/hseditor/hsfonteditor/fonteditor.go

```diff
@@ -32,9 +99,34 @@ func (e *FontEditor) UpdateMainMenuLayout(l *g.Layout) {
 		g.MenuItem("Export to file...").OnClick(func() {}),
 		g.Separator(),
 		g.MenuItem("Close").OnClick(func() {
-			e.Visible = false
+			e.Cleanup()
 		}),
 	})
 
 	*l = append(*l, m)
 }
+
+func (e *FontEditor) GenerateSaveData() []byte {
+	data, err := e.JSON()
+	if err != nil {
+		fmt.Println("failed to marshal font to JSON:, ", err)
+		return nil
+	}
+
+	return data
+}
+
+func (e *FontEditor) Save() {
+	e.Editor.Save(e)
```

Comment:
This is the part I don't like. Due to the way golang does (or, doesn't do) inheritance and overriding, I can't have a generic Save() method in EditorWindow that can just call methods in a child/subclass. I have to do this weird janky stuff. I'm not sure what a more idiomatic solution that doesn't create a bunch of duplicate code would be